src/crimson/net/ProtocolV2.cc

```diff
@@ -719,13 +713,19 @@ seastar::future<bool> ProtocolV2::client_connect()
           if (!server_ident.addrs().contains(conn.target_addr)) {
             logger().warn("{} peer identifies as {}, does not include {}",
                           conn, server_ident.addrs(), conn.target_addr);
-            abort_in_fault();
+            throw std::system_error(
+                make_error_code(ceph::net::error::bad_peer_address));
           }
 
           server_cookie = server_ident.cookie();
 
           // TODO: change peer_addr to entity_addrvec_t
-          ceph_assert(conn.peer_addr == server_ident.addrs().front());
+          if (server_ident.addrs().front() != conn.peer_addr) {
```

Comment:
is this always true? i am afraid that there is chance that the address we picked is not listed in the front.

Reply:
There is change to go wrong, but it will be fixed when peer_addr is changed to `entity_addrvec_t`. Currently our use cases are simple, tests with async-msgr won't use multiple addresses, and with vstart, the front address is always the picked address.